pkg/ds/daemon_set.go

```diff
@@ -54,49 +62,51 @@ type DaemonSet interface {
 
 	// CurrentPods() returns all nodes that are scheduled by this daemon set
 	CurrentPods() (types.PodLocations, error)
-}
 
-// These methods are the same as the methods of the same name in kp.Store.
-// This interface makes writing unit tests easier as daemon sets do not
-// use any functions not listed here
-type kpStore interface {
-	SetPod(
-		podPrefix kp.PodPrefix,
-		nodeName types.NodeName,
-		manifest manifest.Manifest,
-	) (time.Duration, error)
-
-	DeletePod(podPrefix kp.PodPrefix,
-		nodeName types.NodeName,
-		manifestID types.PodID,
-	) (time.Duration, error)
+	// Schedules pods by using replication, this will automatically
+	// cancel any current replications and re-enact
+	PublishToReplication() error
 }
 
 type daemonSet struct {
 	fields.DaemonSet
 
-	logger     logging.Logger
-	kpStore    kpStore
-	scheduler  scheduler.Scheduler
-	dsStore    dsstore.Store
-	applicator labels.Applicator
+	contention    dsContention
+	logger        logging.Logger
+	kpStore       kp.Store
+	scheduler     scheduler.Scheduler
+	dsStore       dsstore.Store
+	applicator    labels.Applicator
+	healthChecker *checker.ConsulHealthChecker
+
+	currentReplication replication.Replication
```

Comment:
comments on these variables would be useful, particularly that `replicationLock` is used to synchronize access to `currentReplication and `replicationOngoing`